src/components/settings/SafeAppsSigningMethod/index.tsx

```diff
@@ -0,0 +1,52 @@
+import ExternalLink from '@/components/common/ExternalLink'
+import { SETTINGS_EVENTS, trackEvent } from '@/services/analytics'
+import { useAppDispatch, useAppSelector } from '@/store'
+import { selectUseOnChainSigning, setUseOnChainSigning } from '@/store/settingsSlice'
+import { FormControlLabel, Checkbox, Paper, Typography, FormGroup, Grid } from '@mui/material'
+
+export const SafeAppsSigningMethod = () => {
+  const useOnChainSigning = useAppSelector(selectUseOnChainSigning)
```

Comment:
I would also suggest to rename the variables that read from redux to not cause confusion with hooks